pricing.js

```diff
@@ -73,11 +73,36 @@ function calculateProductPrice(product, employee, selectedOptions) {
   }
 }
 
+function calculateCommuterPricePerType(type, costs) {
+  const eeCost = costs.find(cost => {
+    return cost.type === type
+  })
+  return eeCost.price
+}
+
+function calculateCommuterPrice(commute, selectedOptions) {
+
+  switch (selectedOptions.benefit) {
+    case 'train':
+    case 'parking':
+      break
+    default:
+      throw new Error(`Unknown commute type: ${selectedOptions.benefit}`)
+  }
+
+  let price
+  let employerContribution
+  price = calculateCommuterPricePerType(selectedOptions.benefit, commute.costs)
+  employerContribution = commute.employerContribution.contribution
```

Comment:
This function, like the others should be responsible for calculating the price of the product. It is the job of `calculateProductPrice` to handle the employer contribution. You never wired this function up to `calculateProductPrice` so in production this code is unreachable.